onyo/lib/onyo.py

```diff
@@ -326,6 +351,15 @@ def add(self, targets: Union[Iterable[Union[Path, str]], Path, str]) -> None:
                 raise FileNotFoundError(f"'{t}' does not exist.")
 
         self._git(['add'] + tgts)
+        # `Repo.add()` is used by most repo-changing commands, and it might be
```

Comment:
Reflecting on this further... I wonder if the cache should not be cleared with `add`. These changes have only been staged, and are not in-effect yet. `commit`, however, that indeed is a change to the onyo repo...

I think what is really exposed here is that there are no docstrings for the properties `files`, `assets`, etc. And thus it's not defined /what/ those properties are guaranteeing. And thus it is not entirely clear when those caches should be cleared.

Reply:
I would say `commit` is a change to a `git` repository, but when you `Repo.add()` something, it is already a change to the onyo repo. I don't want to diverge too much between the too, but I shared my opinion for discussion in the chat to discuss this.

Reply:
We discussed this over a beer last night. I have come to the conclusion that using `Repo`'s public APIs will guarantee cache correctness. So it's about use rather than what's on disk or committed.